test/e2e/federated-ingress.go

```diff
@@ -268,12 +331,23 @@ func waitForIngressShardsGoneOrFail(namespace string, ingress *v1beta1.Ingress,
 	}
 }
 
-func deleteIngressOrFail(clientset *fedclientset.Clientset, namespace string, ingressName string) {
+func deleteIngressOrFail(clientset *fedclientset.Clientset, namespace string, ingressName string, orphanDependents *bool) {
 	if clientset == nil || len(namespace) == 0 || len(ingressName) == 0 {
 		Fail(fmt.Sprintf("Internal error: invalid parameters passed to deleteIngressOrFail: clientset: %v, namespace: %v, ingress: %v", clientset, namespace, ingressName))
 	}
-	err := clientset.Ingresses(namespace).Delete(ingressName, v1.NewDeleteOptions(0))
+	err := clientset.Ingresses(namespace).Delete(ingressName, &v1.DeleteOptions{OrphanDependents: orphanDependents})
 	framework.ExpectNoError(err, "Error deleting ingress %q from namespace %q", ingressName, namespace)
+	// Wait for the ingress to be deleted.
+	err = wait.Poll(5*time.Second, wait.ForeverTestTimeout, func() (bool, error) {
+		_, err := clientset.Extensions().Ingresses(namespace).Get(ingressName)
+		if err != nil && errors.IsNotFound(err) {
```

Comment:
You mean we should still return `false, nil` even if err != nil?
I dont think that we should do that. If there was an error, we should return `false, nil` (except when the error was NotFound, which is what we want and hence return true)